test/cluster/TypeDBClusterServerRunner.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (C) 2021 Vaticle
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package com.vaticle.typedb.common.test.cluster;
+
+import com.vaticle.typedb.common.conf.cluster.Addresses;
+import com.vaticle.typedb.common.test.RunnerUtil;
+import com.vaticle.typedb.common.test.TypeDBRunner;
+import org.zeroturnaround.exec.ProcessExecutor;
+import org.zeroturnaround.exec.ProcessResult;
+import org.zeroturnaround.exec.StartedProcess;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static com.vaticle.typedb.common.test.RunnerUtil.SERVER_STARTUP_TIMEOUT_MILLIS;
+import static com.vaticle.typedb.common.test.RunnerUtil.createProcessExecutor;
+
+public interface TypeDBClusterServerRunner extends TypeDBRunner {
+
+    Addresses address();
+
+    class Factory {
+
+        protected TypeDBClusterServerRunner createServerRunner(Map<String, String> options) {
+            try {
+                return new Default(options);
+            } catch (InterruptedException | TimeoutException | IOException e) {
+                throw new RuntimeException("Unable to construct runner.");
+            }
+        }
+    }
+
+    class Default implements TypeDBClusterServerRunner {
```

Comment:
can this be private now that it should be only be accessed/constructed via the `Factory`?

Reply:
also I've forgotten some context but `Default` seems a bit generic... is this the `MultiProcess` runner as opposed to the `SingleProcess` runner or something?

Reply:
Oh that's a good suggestion actually!

As for making it private, we can't, because the user may still want to access the individual "cluster server" (to stop or restart it, for example).

Reply:
Fixed in 64a90370b223ac489441b51cc64e672e389c097a